bin/varnishd/mgt/mgt_child.c

```diff
@@ -248,16 +248,31 @@ child_listener(const struct vev *e, int what)
 static int __match_proto__(vev_cb_f)
 child_poker(const struct vev *e, int what)
 {
+	char * r = NULL;
+	unsigned status;
 
 	(void)e;
 	(void)what;
 	if (child_state != CH_RUNNING)
 		return (1);
 	if (child_pid < 0)
 		return (0);
-	if (!mgt_cli_askchild(NULL, NULL, "ping\n"))
-		return (0);
-	return (0);
+	if (!mgt_cli_askchild(&status, &r, "ping\n")) {
+		if (strncmp("PONG ", r, 5)) {
+			MGT_complain(C_ERR,
+			    "Unexpected reply from ping: %u %s",
+			    status, r);
+			MGT_Child_Cli_Fail();
+		}
+	}
+	else {
+		MGT_complain(C_ERR, "Did not get a 200 from ping: %u %s",
+		    status, r);
```

Comment:
From a debugging perspective, I think the cases are different enough to produce slightly different errors. But I guess it is a matter of taste.